### ISSUE TITLE
ModuleNotFoundError: No module named 'virtualenv.seed.via_app_data'

### ISSUE THREAD

**Issue by baermathias** [pypa/pipenv#4663]
### Issue description

When running `pipenv install` I get this error: 
```bash
ModuleNotFoundError: No module named 'virtualenv.seed.via_app_data'
✘ Failed creating virtual environment 
```

### Expected result

Create a venv

------------------------------------------------------------------------------

<details><summary>$ pipenv --support</summary>

Pipenv version: `'2020.11.15'`

Pipenv location: `'/home/username/.local/lib/python3.8/site-packages/pipenv'`

Python location: `'/usr/bin/python3'`

Python installations found:

  - `3.8.5`: `/usr/bin/python3`
  - `3.8.5`: `/usr/bin/python3.8`
  - `3.8.5`: `/bin/python3`
  - `3.8.5`: `/bin/python3.8`

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '3.8.5',
 'os_name': 'posix',
 'platform_machine': 'x86_64',
 'platform_python_implementation': 'CPython',
 'platform_release': '5.8.0-45-generic',
 'platform_system': 'Linux',
 'platform_version': '#51~20.04.1-Ubuntu SMP Tue Feb 23 13:46:31 UTC 2021',
 'python_full_version': '3.8.5',
 'python_version': '3.8',
 'sys_platform': 'linux'}
```

System environment variables:

  - `SHELL`
  - `SESSION_MANAGER`
  - `QT_ACCESSIBILITY`
  - `COLORTERM`
  - `XDG_CONFIG_DIRS`
  - `NVM_INC`
  - `XDG_MENU_PREFIX`
  - `GNOME_DESKTOP_SESSION_ID`
  - `MANDATORY_PATH`
  - `LC_ADDRESS`
  - `JAVA_HOME`
  - `GNOME_SHELL_SESSION_MODE`
  - `LC_NAME`
  - `SSH_AUTH_SOCK`
  - `XMODIFIERS`
  - `DESKTOP_SESSION`
  - `LC_MONETARY`
  - `SSH_AGENT_PID`
  - `GTK_MODULES`
  - `DBUS_STARTER_BUS_TYPE`
  - `PWD`
  - `LOGNAME`
  - `XDG_SESSION_DESKTOP`
  - `XDG_SESSION_TYPE`
  - `GPG_AGENT_INFO`
  - `XAUTHORITY`
  - `WINDOWPATH`
  - `HOME`
  - `USERNAME`
  - `IM_CONFIG_PHASE`
  - `LC_PAPER`
  - `LANG`
  - `LS_COLORS`
  - `XDG_CURRENT_DESKTOP`
  - `VTE_VERSION`
  - `INVOCATION_ID`
  - `MANAGERPID`
  - `NVM_DIR`
  - `LESSCLOSE`
  - `XDG_SESSION_CLASS`
  - `ANDROID_HOME`
  - `TERM`
  - `LC_IDENTIFICATION`
  - `DEFAULTS_PATH`
  - `LESSOPEN`
  - `USER`
  - `DISPLAY`
  - `SHLVL`
  - `NVM_CD_FLAGS`
  - `LC_TELEPHONE`
  - `ANDROID_SDK_ROOT`
  - `MOZ_USE_XINPUT2`
  - `QT_IM_MODULE`
  - `LC_MEASUREMENT`
  - `DBUS_STARTER_ADDRESS`
  - `TILIX_ID`
  - `XDG_RUNTIME_DIR`
  - `LC_TIME`
  - `JOURNAL_STREAM`
  - `XDG_DATA_DIRS`
  - `PATH`
  - `GDMSESSION`
  - `DBUS_SESSION_BUS_ADDRESS`
  - `NVM_BIN`
  - `LC_NUMERIC`
  - `OLDPWD`
  - `_`
  - `PIP_DISABLE_PIP_VERSION_CHECK`
  - `PYTHONDONTWRITEBYTECODE`
  - `PIP_SHIMS_BASE_MODULE`
  - `PIP_PYTHON_PATH`
  - `PYTHONFINDER_IGNORE_UNSUPPORTED`

Pipenv–specific environment variables:

Debug–specific environment variables:

  - `PATH`: `/home/username/.local/bin:/home/username/local/bin:/home/username/.nvm/versions/node/v14.15.4/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/local/android/sdkcmdline-tools/tools:/usr/local/android/sdkplatform-tools`
  - `SHELL`: `/bin/bash`
  - `LANG`: `en_US.UTF-8`
  - `PWD`: `/home/username/Documents/business/projects/test/selenium`

---------------------------

</details>

**Comment by frostming** [comment]
What is the version of virtualenv? It can be told by `pip list`

**Comment by gertvdijk** [comment]
I'm also affected by this error and it appears that Pipenv calls an outdated import path, as `virtualenv.seed.via_app_data` is moved to `virtualenv.seed.embed.via_app_data` in https://github.com/pypa/virtualenv/commit/0cd009b5a1338f66397f71c85a75f576a2f3eabf (included since virtualenv 20.0.24).

This is becoming particularly problematic as downgrading virtualenv (e.g. using OS provided one such as on Ubuntu 20.04 ships with virtualenv in /usr/lib/python3/dist-packages version 20.0.17) results in other problems such as outdated embedded `setuptools` being installed in the virtualenv rendering packages uninstallable (e.g. `AttributeError: install_layout` when installing configargparse 1.4 with pipenv).

My work-around for now is to downgrade virtualenv and manually upgrade (pin) all virtualenv-bundled packages (such as setuptools) in my project, which is not very pretty. 

Please include compatibility for virtualenv >= 20.0.24.

**Comment by matteius** [comment]
I believe this can be closed @gertvdijk, but let me know if you are awaiting additional resolutions.
